gnocchi/rest/__init__.py

```diff
@@ -1497,6 +1497,65 @@ def post(self):
 
         pecan.response.status = 202
 
+    @pecan.expose('json')
+    def get_all(self, **kwargs):
+        # Check RBAC policy
+        metric_ids = arg_to_list(kwargs.get('metric', []))
+        metrics = pecan.request.indexer.list_metrics(ids=metric_ids)
+        missing_metric_ids = (set(metric_ids)
+                              - set(six.text_type(m.id) for m in metrics))
+        if missing_metric_ids:
+            # Return one of the missing one in the error
+            abort(404, storage.MetricDoesNotExist(
+                missing_metric_ids.pop()))
+
+        for metric in metrics:
+            enforce("get metric", metric)
+
+        start = kwargs.get('start')
+        if start is not None:
+            try:
+                start = utils.to_datetime(start)
+            except Exception:
+                abort(400, "Invalid value for start")
+
+        stop = kwargs.get('stop')
+        if stop is not None:
+            try:
+                stop = utils.to_datetime(stop)
+            except Exception:
+                abort(400, "Invalid value for stop")
+
+        aggregation = kwargs.get('aggregation', 'mean')
```

Comment:
you could just write that in the signature of the function with `aggregation='mean'` by default. There's too much kwargs going here on. :)